aws/iam.go

```diff
@@ -0,0 +1,59 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+// List all IAM users in the AWS account and returns a slice of the UserNames
+// TODO: Implement exclusion by time filter
+func getAllIamUsers(session *session.Session, region string) ([]*string, error) {
+	svc := iam.New(session)
+	input := &iam.ListUsersInput{}
+
+	var userNames []*string
+
+	// TODO: Probably use ListUsers together with ListUsersPages in case there are lots of users
+	output, err := svc.ListUsers(input)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	for _, user := range output.Users {
+		userNames = append(userNames, user.UserName)
+	}
+
+	return userNames, nil
+}
+
+// Delete all IAM Users
+func nukeAllIamUsers(session *session.Session, userNames []*string) error {
+	if len(userNames) == 0 {
+		logging.Logger.Info("No IAM Users to nuke")
+		return nil
+	}
+
+	logging.Logger.Info("Deleting all IAM Users")
+
+	deletedUsers := 0
+	svc := iam.New(session)
+
+	for _, userName := range userNames {
+		input := &iam.DeleteUserInput{
+			UserName: userName,
+		}
+
+		_, err := svc.DeleteUser(input)
+		if err != nil {
+			logging.Logger.Errorf("[Failed] %s", err)
```

Comment:
We're not consistent about what should happen here, whether to bubble errors or swallow them. I think we are ok with swallowing them for now, but we should address this at some point.

I filed https://github.com/gruntwork-io/cloud-nuke/issues/162 so we have this point in the issue tracker.